### PR TITLE
Zhuqian's assignment 2

### PR DESCRIPTION
**Pull request by KarenZhuqianZhou**
Hi,

Here is my assignment 2. Sorry for the delay. My R crashed down for too many times yesterday... Saaaad...

Luckily, I got them done this morning. But there is a problem with betweenness centrality in my work as you can see at the end of my R markdown (or the html knitted).

Could you please help me with that?

### CODE REVIEW

File: hudk4050-classes.Rmd [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+## Centriality
+
+Lastly, I calculate degree and betweenness centrality respectively utilizing Function degree() and betweenness() and try to find out who is the most central person in the network by showing the first 10 students with the highest centrality.
+
+```{r}
+degree.student <- degree(C10,v=V(C10), mode = "all", loops = TRUE, normalized = FALSE)
+names(degree.student) <- V(C10)$name
+ind <- order(-degree.student)
+degree.student[ind][1:10]
+btwn.student <- betweenness(C10, v = V(C10), directed = FALSE, nobigint = TRUE, normalized = TRUE)
```
Comment (charles-lang) [inline hudk4050-classes.Rmd]: Did you try "normalized = FALSE"? It may be that in the normalization calculation something is being divided by zero?

Comment (KarenZhuqianZhou) [inline hudk4050-classes.Rmd]: Thanks! As for documenting, I am just afraid of forgetting how I managed it this time ^_~ .
And I've tried both "TRUE" and "FALSE" but failed. I will try to figure it out later. Maybe it's a reminder for me to ask a question on Stack Overflow.

**Comment by charles-lang** [comment]
Hmmm, very strange. I’m sure you can figure it out though. Perhaps it is because there are so many nodes with no connections. You could try to calculate the betweenness after you remove all the students who have not connections?

On Sat, Oct 15, 2016 at 2:00 PM KarenZhuqianZhou

<
mailto:KarenZhuqianZhou <notifications@github.com>
> wrote:

a, pre, code, a:link, body { word-wrap: break-word !important; }

@KarenZhuqianZhou

commented on this pull request.

In
https://github.com/core-methods-in-edm/assignment2/pull/20
:

> +```{r Network of Students, echo=TRUE} +library(igraph) +C10 <- graph.adjacency(C9, mode="undirected", diag=FALSE) +plot(C10, layout = layout.fruchterman.reingold, vertex.label=NA, vertex.size = 5, edge.width = get.edge.attribute(C10, "weight")) +``` + +## Centriality + +Lastly, I calculate degree and betweenness centrality respectively utilizing Function degree() and betweenness() and try to find out who is the most central person in the network by showing the first 10 students with the highest centrality. + +```{r} +degree.student <- degree(C10,v=V(C10), mode = "all", loops = TRUE, normalized = FALSE) +names(degree.student) <- V(C10)$name +ind <- order(-degree.student) +degree.student[ind][1:10] +btwn.student <- betweenness(C10, v = V(C10), directed = FALSE, nobigint = TRUE, normalized = TRUE)

Thanks! As for documenting, I am just afraid of forgetting how I managed it this time ^_~ .

And I've tried both "TRUE" and "FALSE" but failed. I will try to figure it out later. Maybe it's a reminder for me to ask a question on Stack Overflow.

—

You are receiving this because you commented.

Reply to this email directly,
https://github.com/core-methods-in-edm/assignment2/pull/20
, or

**Comment by KarenZhuqianZhou** [comment]
Hmm, sounds a great idea!

I've tried to use a subset that contains just three students and two of
them are connected, but still all zero. Since there is also an isolated
person among the three, it may be the problem. I will try to calculate btwn
of a subset that all are connected to one another. Hope it will work then.

**Pull request closed without merging**